drivers/include/mtd.h

```diff
@@ -507,16 +507,27 @@ int mtd_erase(mtd_dev_t *mtd, uint32_t addr, uint32_t count);
 int mtd_erase_sector(mtd_dev_t *mtd, uint32_t sector, uint32_t num);
 
 /**
- * @brief   Check whether a sector needs to be erased before it can be written
+ * @brief   Write data to a MTD device with whole sector writes
  *
- * @param      mtd    the device to erase
- * @return true if an erase is needed, false if old data will be
- *         overwritten automatically.
+ * The MTD layer will take care of splitting up the transaction into multiple
+ * writes if it is required by the underlying storage media.
+ *
+ * The sectors will be erased before writing if needed.
+ *
+ * @param      mtd      the device to write to
+ * @param[in]  src      the buffer to write
+ * @param[in]  sector   Sector number to start writing to
+ * @param[in]  size     the number of sectors to write
```

Comment:
In all other parameter descriptions we don't use `the`
```suggestion
 * @param      mtd      Device to write to
 * @param[in]  src      Buffer to write
 * @param[in]  sector   Sector number to start writing to
 * @param[in]  size     Number of sectors to write
```
It would be more consistent, especially with the `sector` paraemeter. You might squash it directly.